### PR TITLE
Fixed SourceText.Write for spans that don't start at 0 

### LINKED ISSUE #41903: SourceText.Write(TextWriter, TextSpan, CancellationToken) is unusable with a span start other than 0
**Version Used**: 3.4.0 nupkg

For each character excluded from the start of the text by the start of the passed span, a character is incorrectly additionally excluded from the end of the text.

```cs
using Microsoft.CodeAnalysis.Text;
using System;

public static class Program
{
    public static void Main()
    {
        var sourceText = SourceText.From("ABCDEFGHIJKLMNOPQRSTUVWXYZ");

        // Should print B to Z, but prints BCDEFGHIJKLMNOPQRSTUVWXY
        sourceText.Write(Console.Out, TextSpan.FromBounds(1, sourceText.Length));
        Console.WriteLine();

        // Should print K to Z, but prints KLMNOP
        sourceText.Write(Console.Out, TextSpan.FromBounds(10, sourceText.Length));
        Console.WriteLine();

        // Is an invalid span and SourceText.Write throws (as it should)
        sourceText.Write(Console.Out, TextSpan.FromBounds(1, sourceText.Length + 1));

        // Is an invalid span and SourceText.Write throws (as it should)
        sourceText.Write(Console.Out, TextSpan.FromBounds(10, sourceText.Length + 1));
    }
}
```

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net48</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.CodeAnalysis" Version="3.4.0" />
  </ItemGroup>

</Project>
```

### PR DESCRIPTION
**Pull request by jnm2**
Fixes #41903. Three simplification commits follow.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@cston Do we need a second compiler review, or is it enough for the two of us to do it?

**Comment by cston** [comment]
@CyrusNajmabadi, we should get a second compiler review since it's a compiler change.

**Comment by CyrusNajmabadi** [comment]
SGTM.  Wasn't sure what policy was.  @dotnet/roslyn-compiler can someone PTAL at this community PR?  Thanks!

File: src/Compilers/Core/Portable/Text/SourceText.cs [hunk 1]
```diff
@@ -469,7 +469,8 @@ public virtual SourceTextContainer Container
 
         internal void CheckSubSpan(TextSpan span)
         {
-            if (span.Start < 0 || span.Start > this.Length || span.End > this.Length)
+            // TextSpan guarantees that Start >= 0 and that Start <= End.
```
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/Text/SourceText.cs]: >// TextSpan guarantees that Start >= 0 and that Start <= End. [](start = 12, length = 61)

We usually use asserts to document and enforce assumptions like this.

Comment (jnm2) [reply src/Compilers/Core/Portable/Text/SourceText.cs]: Thanks, done!

**Comment by cston** [comment]
@jnm2, thanks for the contribution.

**Comment by jnm2** [comment]
Thanks for the timely merge!

**Pull request merged**